flight/PiOS/STM32F4xx/pios_dmashot.c

```diff
@@ -0,0 +1,591 @@
+/**
+ ******************************************************************************
+ * @file       pios_dmashot.c
+ * @author     dRonin, http://dRonin.org/, Copyright (C) 2017
+ * @addtogroup PIOS PIOS Core hardware abstraction layer
+ * @{
+ * @addtogroup PIOS_DMAShot PiOS DMA-driven DShot driver
+ * @{
+ * @brief Generates DShot signal by updating timer CC registers via DMA bursts.
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, see <http://www.gnu.org/licenses/>
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+#include <pios.h>
+#include "pios_dmashot.h"
+
+#define MAX_TIMERS                              8
+
+// This is to do half-word writes where appropriate. TIM2 and TIM5 are 32-bit, the rest
+// are 16-bit. Can do full word writes to 16bit regardless, but it doubles DMA buffer size
+// unnecessarily.
```

Comment:
Just checked again, doesn't work. RE1 has pin 1-4 on TIM5, which is 32bit. When I try to do half-word DMA into the DMAR of a 32bit timer, it doesn't seem to execute the transfer.